Project/ComponentMesh.cpp

```diff
@@ -26,50 +26,52 @@ void ComponentMesh::OnEditorUpdate()
 {
 	GameObject* selected = App->editor->panel_hierarchy.selected_object;
 	std::vector<ComponentMesh*> meshes = selected->GetComponents<ComponentMesh>();
+	int count = 1;
+
 	for (ComponentMesh* mesh : meshes)
 	{
-		int count = 1;
+		// Show only # when multiple
 		char name[50];
-		sprintf_s(name, 50, "Mesh %d", count);
-		if (mesh != nullptr)
+		if (meshes.size() == 1)
 		{
-			// Show only # when multiple
-			if (meshes.size() == 1)
-			{
-				sprintf_s(name, 50, "Mesh");
-			}
-			if (ImGui::CollapsingHeader(name))
+			sprintf_s(name, 50, "Mesh");
+		}
+		else
+		{
+			sprintf_s(name, 50, "Mesh %d", count);
+		}
+
+		if (ImGui::CollapsingHeader(name))
+		{
+			bool active = this->IsActive();
+			if (ImGui::Checkbox("Active##mesh", &active))
 			{
-				bool active = this->IsActive();
-				if (ImGui::Checkbox("Active##mesh", &active))
+				if (active)
 				{
-					if (active)
-					{
-						this->Enable();
-					}
-					else
-					{
-						this->Disable();
-					}
+					this->Enable();
 				}
-				ImGui::SameLine();
-				if (ImGui::Button("Remove##mesh"))
+				else
 				{
-					// TODO: Add delete Component tool
+					this->Disable();

```

Comment:
No necesitas el this->.

Reply:
Cambiando a `active ? Enable() : Disable();`